lib/aws-record/record/item_operations.rb

```diff
@@ -641,10 +633,22 @@ def _build_update_expression(attr_value_pairs)
           update_expressions = []
           update_expressions << ("SET #{set_expressions.join(', ')}") unless set_expressions.empty?
           update_expressions << ("REMOVE #{remove_expressions.join(', ')}") unless remove_expressions.empty?
-          if update_expressions.empty?
-            nil
-          else
-            [update_expressions.join(' '), exp_attr_names, exp_attr_values]
+          {
+            update_expression: update_expressions.join(' '),
+            expression_attribute_names: exp_attr_names,
+            expression_attribute_values: exp_attr_values
+          }.reject { |_, value| value.nil? || value.empty? }
+        end
+
+        def _merge_update_expression_opts(update_expression_opts, pass_through_opts)
+          update_expression_opts.merge(pass_through_opts) do |key, expression_value, pass_through_value|
+            case key
+            when :update_expression
+              msg = 'Using pass-through update expression with attribute updates is not currently supported.'
```

Comment:
I think saying "not currently supported" implies a future support, which I dont think we would do: it seems to be like a invalid input to specify both and that should always be an error.  Maybe update this to say its a conflict?

Reply:
Fair! I was mostly following the example of [this language](https://github.com/lsglick/aws-sdk-ruby-record/blob/888e5bf0cb62f3b8355acc0bad0117513bb9bdba/lib/aws-record/record/transactions.rb#L256) and selfishly hopeful there might be a way to merge the full update expression (e.g. add `ADD` or additional `SET` clauses to the expression generated by attribute updates) as long as individual attributes don't conflict.

But agree not accurate to imply this _will_ happen.